src/PotentialCalculation/SuccessiveOverRelaxation/SOR.jl

```diff
@@ -71,33 +72,58 @@ Thus, there cannot be local extrema.
 """
 @inline @fastmath function handle_depletion(
     new_potential::T, 
-    point_type::PointType, 
+    imp_scale::T,
     neighbor_potentials::NTuple{6,T}, 
     q_eff_imp::T, 
     volume_weight::T,
     sor_const::T
-)::Tuple{T, PointType} where {T}
+)::Tuple{T, T} where {T}
     vmin::T = min(neighbor_potentials...)
     vmax::T = max(neighbor_potentials...)
+    
+    isLess = new_potential < vmin
+    neighbor_relevant_extremum = isLess ? vmin : vmax
 
-    if new_potential < vmin || new_potential > vmax
-        new_potential -= q_eff_imp * volume_weight * sor_const
-        if (point_type & undepleted_bit == 0 && point_type & pn_junction_bit > 0) 
-            point_type += undepleted_bit
+    Δnp_ext = new_potential - neighbor_relevant_extremum
+    imp_contribution_fix = q_eff_imp * volume_weight * sor_const 
+    imp_contribution_current = imp_contribution_fix * imp_scale
+    
+    if isLess || new_potential > vmax
+        new_potential = neighbor_relevant_extremum 
+        if imp_contribution_current != 0 
+            x::T = imp_contribution_current - Δnp_ext
+            new_imp_scale::T = x / imp_contribution_fix
+            if new_imp_scale > 1
+                new_imp_scale = 1
+            end
+            imp_scale::T = new_imp_scale
         end
-    elseif point_type & undepleted_bit > 0
-        point_type -= undepleted_bit
+    else
+        imp_scale *= 1.1 # Just chosen. Seems to work. There might be a better solution.
```

Comment:
Please elaborate on this.

Reply:
At the edge of the depleted zone, `imp_scale` should converge against some value `x` in `[0, 1]`. 
The `else`-part should be to increase it again a little bit (towards `x`) in case it was reduced to some value smaller than `x`
in the previous iteration.